libr/search/search.c

```diff
@@ -439,11 +360,11 @@ R_API void r_search_kw_reset(RSearch *s) {
 }
 
 R_API void r_search_reset(RSearch *s, int mode) {
+	R_FREE (s->data);
```

Comment:
The check was not there before my change. Do that in another commit if you really care.